modules/iknowpy/iknowpy/engine.pyx

```diff
@@ -41,75 +43,113 @@ cdef char* aType_to_str(Attribute t) except NULL:
 	raise ValueError('Attribute type {} is unrecognized.'.format(t))
 
 
-cdef class UserDictionary:
+Entry = namedtuple("Entry", ["literal", "label"])
+
+cdef class UserDictionary(object):
 	"""A class that represents a user dictionary"""
 	cdef CPPUserDictionary user_dictionary
 
-	def __cinit__(self):
-		"""Initialize the underlying C++ iKnowUserDictioanry class"""
+	# keep track of entries in a readable form
+	cdef public entries
```

Comment:
The `entries` attribute should be made read-only so that users of this class cannot modify it directly. To do this, you can name the attribute `_entries`. Then use the `@property` decorator to create a getter named `entries`. I would also recommend that you return a copy of the list in the getter so that the user cannot directly mutate the list. Something like this should work:

```
cdef public list _entries

@property
@cython.binding(True)
def entries(self):
    return self._entries[:]
```